extensions/big-data-cluster/src/common/promise.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Deferred promise
+ */
+export class Deferred<T> {
```

Comment:
is this really needed?

Reply:
Yeah I use it in https://github.com/microsoft/azuredatastudio/pull/7784/files#diff-8a0ab0c276672055ff30af1c7adf0cd1R117